### PR TITLE
Change paging model name to ListResult

### PR DESCRIPTION
**Pull request by pshao25**
Fix https://github.com/Azure/autorest.csharp/issues/2989

### CODE REVIEW

File: src/CADL.Extension/Emitter.Csharp/src/lib/templateHandler.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+/**
+ * Pagination model should have name {ResourceType}ListResult
+ */
+export function isPagingModel(type: Type) : boolean {
+    return type.kind === "Model" && hasDecorator(type, "$pagedResult");
+}
+
+export function getPagingModelName(model: Model) : string | undefined {
```
Comment (lirenhe) [inline src/CADL.Extension/Emitter.Csharp/src/lib/templateHandler.ts:24]: Will other SDKs also need this? How do you think of putting it in Cadl-dpg?

File: src/CADL.Extension/Emitter.Csharp/src/lib/templateHandler.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    if (isPagingModel(model)) {
+        if (!model.templateArguments || model.templateArguments[0].kind !== "Model") {
+            throw "Paging model should have a containing model resource";
+        }
+        
+        return `${model.templateArguments[0].name}ListResult`;
```
Comment (lirenhe) [inline src/CADL.Extension/Emitter.Csharp/src/lib/templateHandler.ts:30]: Does this name (nameListResult) align with the naming of current HLC or mgmt. SDK?

File: src/CADL.Extension/Emitter.Csharp/src/lib/templateHandler.ts (hunk 1 above)
Comment (pshao25) [reply src/CADL.Extension/Emitter.Csharp/src/lib/templateHandler.ts:24]: Others don't need it. It is .net specific.

File: src/CADL.Extension/Emitter.Csharp/src/lib/templateHandler.ts (hunk 2 above)
Comment (pshao25) [reply src/CADL.Extension/Emitter.Csharp/src/lib/templateHandler.ts:30]: Yes, we are following the current naming convention.

**Comment by pshao25** [comment]
Don't need that now.

**Pull request closed without merging**